src/resolvers/git.cr

```diff
@@ -103,9 +103,19 @@ module Shards
     def self.normalize_key_source(key : String, source : String) : {String, String}
       case key
       when "git"
-        {"git", source}
+        uri = URI.parse(source)
+        downcased_host = uri.host.try &.downcase
+        if downcased_host && downcased_host.in?("github.com", "bitbucket.com", "gitlab.com")
+          uri.scheme = "https"
```

Comment:
I'm not sure changing schemes is a good idea.

There could be a reason you need to access via http protocol instead of https. I guess it doesn't matter much though because in practice those are just redirects.

Reply:
Yeah, but if user depends on invalid `http` based path it should raise instead of silently change it to the working version.

Reply:
Let's keep the protocol so we don't prevent the user to use `http://` for whatever reason and play it safe for now in that regard.

The initial issue is the owner/repo name, so it's still an improvement.

Reply:
I've limited the protocol changes so people can use `ssh` for private repos

Reply:
remembering that these services are known quantities and you can't actually clone over HTTP

Reply:
> I've limited the protocol changes so people can use `ssh` for private repos

```suggestion
```

It's still there.

Reply:
This will make impossible to use `git` protocol, I'm not sure why would you like to forbid that?